Remove deprecated use of Pydantic

The latest version of Pydantic, 2.11, is causing errors such as:

```console
=========================== short test summary info ============================
FAILED dandi/cli/tests/test_cmd_ls.py::test_ls_bids_file - KeyError: 'identifier'
FAILED dandi/cli/tests/test_cmd_ls.py::test_ls_zarrbids_file - KeyError: 'identifier'
FAILED dandi/cli/tests/test_cmd_ls.py::test_smoke_local_schema - KeyError: 'digest'
FAILED dandi/cli/tests/test_cmd_validate.py::test_validate_bids_error[invalid_asl003] - AssertionError: assert 'NON_BIDS_PATH_PLACEHOLDER' in 'Error: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.\n'
 +  where 'Error: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.\n' = <Result SystemExit(1)>.output
FAILED dandi/cli/tests/test_cmd_validate.py::test_validate_bids_error[invalid_pet001] - AssertionError: assert 'NON_BIDS_PATH_PLACEHOLDER' in 'Error: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.\n'
 +  where 'Error: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.\n' = <Result SystemExit(1)>.output
FAILED dandi/tests/test_metadata.py::test_bids_nwb_metadata_integration - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_metadata.py::test_prepare_metadata[metadata2asset.json-metadata0] - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_metadata.py::test_prepare_metadata[metadata2asset_simple1.json-metadata1] - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_metadata.py::test_prepare_metadata[metadata2asset_3.json-metadata2] - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_metadata.py::test_prepare_metadata[metadata2asset_cellline.json-metadata3] - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_metadata.py::test_nwb2asset - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_upload.py::test_upload_download - dandi.exceptions.UploadError: failed to extract metadata: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
FAILED dandi/tests/test_upload.py::test_upload_bids_invalid - pydantic.warnings.PydanticDeprecatedSince211: Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3. Instead, you should access this attribute from the model class. Deprecated in Pydantic V2.11 to be removed in V3.0.
```

Remove deprecated uses of Pydantic.

See https://github.com/dandi/dandi-cli/actions/runs/14181958193/job/39729884379 for details of the errors.

--- a/dandi/metadata/util.py
+++ b/dandi/metadata/util.py
@@ -515,7 +515,7 @@ M = TypeVar("M", bound=models.DandiBaseModel)
 
 def extract_model(modelcls: type[M], metadata: dict, **kwargs: Any) -> M:
     m = modelcls.model_construct()
-    for field in m.model_fields.keys():
+    for field in modelcls.model_fields.keys():
         value = kwargs.get(field, extract_field(field, metadata))
         if value is not None:
             setattr(m, field, value)

